### PR TITLE
feat: boojum integration

### PR DESCRIPTION
**Pull request by AntonD3**
# What :computer: 
1. System contracts updated to the version currently used on the mainnet
2. ~`zksync-era` update to the last commit of the main branch~
3. The latest vm version being used
4. Added `EcAdd` and `EcMul` precompiles
5.  Account impersonating now works using a specialized bootloader
6. `eth_sendTransaction` supported
7. Added account impersonating to the fee estimation method
8. `etc/system-contracts/SystemConfig.json` file added to track the system contracts version and changes
9.  System contracts building scripts were updated

# Why :hand:
1-4. to have the same state as on the mainnet after the Boojum upgrade
5. It allows us to avoid all the checks, the sender can be a contract, from kernel space, or anything. May be useful for some tests, widely used in the system contracts tests.
6. It allows to use the hardhat tools while impersonating accounts, which is really convenient
7. To have the same behavior as during executing
8. It will be easier to track the SC version, update it.
9. To work with the new system contracts

# Evidence :camera:
Include screenshots, screen recordings, or `console` output here demonstrating that your changes work as intended

### CODE REVIEW

File: etc/system-contracts/bootloader/test_infra/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+[package]
+name = "test_infra"
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+
+vm = { git = "https://github.com/matter-labs/zksync-era.git", branch = "boojum-integration" }
```
Comment (Deniallugo) [inline etc/system-contracts/bootloader/test_infra/Cargo.toml:10]: Could you please reference to the main branch. 
I see that vm crate is ultra outdated

Comment (AntonD3) [reply etc/system-contracts/bootloader/test_infra/Cargo.toml:10]: It was copied from the original system contracts, and it's outdated there. We will update it, but maybe with 1.4.1. However here this script will not be used, and actually, we can remove it.

Comment (MexicanAce) [reply etc/system-contracts/bootloader/test_infra/Cargo.toml:10]: Tried to remove the test directories, but they're needed as part of the `etc/system-contracts/scripts/process.ts` file

Will leave as-is for now

**Review by MexicanAce — APPROVED** [review]
LGTM 🚀

**Pull request merged**